Fix documentation on sharing outputs between projects (#34011)
gradle/gradle: platforms/documentation/docs/src/docs/userguide/dep-man/how-to/how_to_share_outputs_between_projects.adoc

[[sec:setup-consumer]]
== Step 2: Configuring the consumer project

> In Gradle, it's important to separate the configuration where you declare dependencies from the one where you resolve artifacts.

Reviewer: ```suggestion
In Gradle, it's recommended to separate the configuration where you declare dependencies from the one where you resolve artifacts.
```

You could still use a legacy configuration to both declare and resolve (for now).

This avoids ambiguity, improves maintainability, and ensures compatibility with Gradle's evolving configuration model.

First, the consumer project creates two configurations:

> * `instrumentedRuntimeDependencies` is used to declare that the consumer depends on the producer's `instrumentedJars`.

Reviewer 2: ```suggestion
* `instrumentedRuntimeDependencies` is used to declare which producer that the consumer depends on.
```

Author: Done.

> * `instrumentedRuntime` is a resolvable configuration that extends the dependency declaration and is used by the `runWithInstrumentation` task.

Reviewer 2: ```suggestion
* `instrumentedRuntime` is a resolvable configuration that resolves the `instrumentedJars` from the dependencies declared in `instrumentedRuntimeDependencies`.
```

Author: Done.

====
include::sample[dir="snippets/how-to/share-outputs/kotlin",files="consumer/build.gradle.kts[tags=custom-config]"]
include::sample[dir="snippets/how-to/share-outputs/groovy",files="consumer/build.gradle[tags=custom-config]"]
====

> Now, the consumer project requests the instrumented JAR by defining matching attributes.

Author: I updated to:

```suggestion
Now, the consumer project aligns its configuration with the desired variant by defining matching attributes. This ensures that, when the configuration is resolved, Gradle will select the instrumented JAR variant from the producer if it’s available.
```

If the consumer fails to resolve the artifact, check:

> - The attributes in the consumer are compatible to those in the producer.

Reviewer: ```suggestion
- The attributes in the consumer are compatible with those in the producer.
```
